plan/logical_plan_builder.go

```diff
@@ -1139,6 +1153,15 @@ func (b *planBuilder) buildDelete(delete *ast.DeleteStmt) LogicalPlan {
 	if b.err != nil {
 		return nil
 	}
+
+	if ds, ok := p.(*DataSource); ok {
```

Comment:
you should check every `delete tables`, If the delete table is null, you should check every table refs.